src/dmd/dsymbolsem.d

```diff
@@ -1215,9 +1215,26 @@ private extern(C++) final class DsymbolSemanticVisitor : Visitor
                 }
                 else
                 {
+                    import dmd.hdrgen : toCBuffer, HdrGenState;
+                    OutBuffer buff;
+                    HdrGenState hgs;
+                    toCBuffer(dsym, &buff, &hgs);
+                    if (buff.offset != 0)
+                    {
+                        version(Windows)
+                            buff.setsize(buff.offset-2);
+                        else
+                            buff.setsize(buff.offset-1);
+                    }
+
                     // https://issues.dlang.org/show_bug.cgi?id=14166
                     // Don't run CTFE for the temporary variables inside typeof
                     dsym._init = dsym._init.initializerSemantic(sc, dsym.type, sc.intypeof == 1 ? INITnointerpret : INITinterpret);
+                    const init_err = dsym._init.isExpInitializer();
+                    if (init_err && init_err.exp.op == TOK.errExtraInfo)
+                    {
+                         errorSupplemental(dsym.loc, "compile time context created here: `%s`", buff.peekString());
```

Comment:
I think it's somewhat of an abuse to use the `hdrgen` code for this purpose, as that's not what it was intended for.

I don't think it's absolutely necessary to print the statement; the line number should be reasonably sufficient.  There are other places in the compiler where the line number alone does the job:

```
src/dmd/parse.d:4207:                        error("C++ name spaces not allowed here");
src/dmd/parse.d:4211:                        error("C++ mangle declaration not allowed here");
src/dmd/ctfeexpr.d:194:            errorSupplemental(loc, "thrown from here");
src/dmd/dinterpret.d:5404:            errorSupplemental(result.loc, "originally uninitialized here");
```

Reply:
Seconded!
there's no need to use the hdrgen here.

Reply:
Addressed!